privatek8s.tf

```
@@ -1,145 +1,145 @@
-#   scope                            = "${data.azurerm_subscription.jenkins.id}/resourceGroups/${data.azurerm_resource_group.private_prod.name}/providers/Microsoft.Network/virtualNetworks/${data.azurerm_virtual_network.private_prod.name}/subnets/${azurerm_subnet.privatek8s_tier.name}" # azurerm_subnet.privatek8s_tier.name
-#   role_definition_name             = "Network Contributor"
-#   principal_id                     = azurerm_kubernetes_cluster.privatek8s.identity[0].principal_id
-#   skip_service_principal_aad_check = true
-# }
-
-# resource "kubernetes_storage_class" "managed_csi_premium_retain" {
-#   metadata {
-#     name = "managed-csi-premium-retain"
-#   }
-#   storage_provisioner = "disk.csi.azure.com"
-#   reclaim_policy      = "Retain"
-#   parameters = {
-#     skuname = "Premium_LRS"
-#   }
-#   provider = kubernetes.privatek8s
-# }
-
-# output "privatek8s_kube_config" {
-#   value     = azurerm_kubernetes_cluster.privatek8s.kube_config_raw
-#   sensitive = true
-# }
-
-# output "privatek8s_public_ip_address" {
-#   value = azurerm_public_ip.public_privatek8s.ip_address
-# }
+data "azurerm_subscription" "jenkins" {
+}
+
+resource "azurerm_resource_group" "privatek8s" {
+  name     = "prod-privatek8s"
+  location = var.location
+  tags     = local.default_tags
+}
+
+resource "random_pet" "suffix_privatek8s" {
+  # You want to taint this resource in order to get a new pet
+}
+
+# Important: the Enterprise Application "terraform-production" used by this repo pipeline needs to be able to manage this subnet
+# See the corresponding role assignment for this cluster added here (private repo):
+# https://github.com/jenkins-infra/terraform-states/blob/1f44cdb8c6837021b1007fef383207703b0f4d76/azure/main.tf#L49
+resource "azurerm_subnet" "privatek8s_tier" {
+  name                 = "privatek8s-tier"
+  resource_group_name  = data.azurerm_resource_group.prod_private.name
+  virtual_network_name = data.azurerm_virtual_network.prod_private.name
+  address_prefixes     = ["10.244.0.0/16"]
+}
+
+#tfsec:ignore:azure-container-logging #tfsec:ignore:azure-container-limit-authorized-ips
+resource "azurerm_kubernetes_cluster" "privatek8s" {
+  name                              = "privatek8s-${random_pet.suffix_privatek8s.id}"
+  location                          = azurerm_resource_group.privatek8s.location
+  resource_group_name               = azurerm_resource_group.privatek8s.name
+  kubernetes_version                = var.kubernetes_version
+  dns_prefix                        = "privatek8s-${random_pet.suffix_privatek8s.id}"
+  role_based_access_control_enabled = true # default value, added to please tfsec
+  # api_server_authorized_ip_ranges   = ["176.185.227.180/32"] # TODO: set correct value
+
+  network_profile {
+    network_plugin = "azure"
+    network_policy = "azure"
+    ## https://learn.microsoft.com/en-gb/azure/aks/configure-kubenet-dual-stack
+    # ip_versions = ["IPv4", "IPv6"]
```

```suggestion
    ip_versions = ["IPv4", "IPv6"]
```

Since we created the new networks? If this does not work, then we'll have to fix the networks before creating the cluster (because changing this attributes force re-creation of the cluster resources)

---

I intended to do that in another PR. WDYT?

---

Fine for me to activate it right now